File: framework/languages/java/samples/java/ShoppingMall/Client/src/main/java/systems/zlink/samples/shoppingmall/client/configuration/SampleTopology.java
package systems.zlink.samples.shoppingmall.client.configuration;

public final class SampleTopology {
    public static final String ApiAHttpUrl = property("apiAHttpUrl", "http://127.0.0.1:49101");
    public static final String ApiBHttpUrl = property("apiBHttpUrl", "http://127.0.0.1:49102");

    private SampleTopology() {
    }

    private static String property(String name, String fallback) {
        return System.getProperty("zlink.samples.shoppingmall." + name, fallback);
    }
}

File: framework/languages/java/samples/kotlin/ShoppingMall/Client/src/main/kotlin/systems/zlink/samples/kotlin/shoppingmall/client/configuration/SampleTopology.kt
package systems.zlink.samples.kotlin.shoppingmall.client.configuration

object SampleTopology {
    val CommerceApiAEndpoint = property("commerceApiAEndpoint", "tcp://127.0.0.1:47492")
    val CommerceApiBEndpoint = property("commerceApiBEndpoint", "tcp://127.0.0.1:47493")

    fun commerceApiEndpoint(instanceId: String): String =
        if (instanceId == SampleNames.ApiInstanceB) CommerceApiBEndpoint else CommerceApiAEndpoint

    private fun property(name: String, defaultValue: String): String =
        System.getProperty("zlink.samples.shoppingmall.$name", defaultValue)
}
